engine/src/main/java/org/destinationsol/game/Hero.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright 2018 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.destinationsol.game;
+
+import com.badlogic.gdx.math.Vector2;
+import org.destinationsol.common.SolDescriptiveException;
+import org.destinationsol.game.input.Pilot;
+import org.destinationsol.game.item.Armor;
+import org.destinationsol.game.item.ItemContainer;
+import org.destinationsol.game.item.Shield;
+import org.destinationsol.game.item.SolItem;
+import org.destinationsol.game.item.TradeContainer;
+import org.destinationsol.game.ship.FarShip;
+import org.destinationsol.game.ship.ShipAbility;
+import org.destinationsol.game.ship.SolShip;
+import org.destinationsol.game.ship.hulls.Hull;
+
+/**
+ * A wrapper class for the Hero, that handles the normal and transcendent ships transparently.
+ */
+public class Hero {
+    private SolShip hero;
+    private StarPort.Transcendent transcendentHero;
+    private FarShip transcendentHeroShip;
+    private boolean isTranscendent;
+    private boolean isDead;
+
+    public Hero(SolShip hero) {
+        this.hero = hero;
+        if (hero == null) {
+            throw new SolDescriptiveException("Something tries to create hero, when there is no ship linked to hiim. Please report this..");
+        }
+        isTranscendent = false;
+    }
+
+    public void toTranscendent(StarPort.Transcendent transcendentHero) {
```

Comment:
`toXYZ` methods are generally used to specify object conversions. If we choose to follow that convention, it might make sense to rename this method to something like `setTranscendent`.

Reply:
This is kind of object conversion